adm_program/modules/dates/dates_function.php

```diff
@@ -77,12 +70,24 @@
     $date->readDataById($getDateId);
 
     // Pruefung, ob der Termin zur aktuellen Organisation gehoert bzw. global ist
-    if(!$date->editable())
+    if(!$date->editable()
+    && $getMode !== 3 && $getMode !== 4 && $getMode !== 6 && $getMode !== 7)
     {
         $gMessage->show($gL10n->get('SYS_NO_RIGHTS'));
         // => EXIT
     }
 }
+else
+{
+    // check if the user has the right to edit at least one category
+    if(count($gCurrentUser->getAllEditableCategories('DAT')) === 0
+    && $getMode !== 3 && $getMode !== 4 && $getMode !== 6 && $getMode !== 7)
```

Comment:
maybe use `in_array()` with the allowed values

Reply:
done